pandas/tests/indexes/datetimes/test_constructors.py

```diff
@@ -950,3 +950,60 @@ def test_datetimeindex_constructor_misc(self):
         )
         assert len(idx1) == len(idx2)
         assert idx1.freq == idx2.freq
+
```

Comment:
We should also test that conflicting `fold` values raise in the constructor
e.g.
```
Timestamp(datetime(2020, 1, 1, fold=0), fold=1))
```

Reply:
@mroeschke 
Currently this doesn't raise, but if we implement it, then ,because datetime always has fold, this will also raise an error:
```python
Timestamp(datetime(2020, 1, 1), fold=1)
```
Because it's equivalent to:
```python
Timestamp(datetime(2020, 1, 1, fold=0), fold=1))
```
Is this what we want to do? If that's the case, we can just raise if fold is passed with datetime-like `ts-input`.

Reply:
I was thinking if `fold=None` is the default in the `Timestamp` constructor, this should be the following behavior. Do you agree?

```
Timestamp(datetime(2020, 1, 1, fold=0), fold=None) -> fold = 0
Timestamp(datetime(2020, 1, 1, fold=1), fold=None) -> fold = 1
Timestamp(datetime(2020, 1, 1, fold=0), fold=0) -> fold = 0
Timestamp(datetime(2020, 1, 1, fold=1), fold=0) -> raise
Timestamp(datetime(2020, 1, 1, fold=0), fold=1) -> raise
Timestamp(datetime(2020, 1, 1, fold=1), fold=1) -> fold = 1

```

Reply:
Yeah, let's do that. The worry I had was that someone might receive datetime-like data from outside and convert it to Timestamp with fold. Then everything would be fine until there would be a conflict.
But that's a pretty contrived case, and I think that whoever gets into it will be able to pull themselves out.

Reply:
Hmm. **continuous-integration/travis-ci/pr** seems to be failing for everyone.